frontend/src/core/user/components/UserMenu.test.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import SignOut from './SignOut';
+import { UserMenuBase } from './UserMenu';
+
+
+describe('<UserMenuBase>', () => {
+    it('shows the user avatar when the user is logged in', () => {
+        const wrapper = shallow(
+            <UserMenuBase user={ { isAuthenticated: true } }/>
+        );
+
+        expect(wrapper.find('img')).toHaveLength(1);
+        expect(wrapper.find('.menu-icon')).toHaveLength(0);
+    });
+
+    it('shows the general menu icon when the user is logged out', () => {
+        const wrapper = shallow(
+            <UserMenuBase user={ { isAuthenticated: false } }/>
+        );
+
+        expect(wrapper.find('img')).toHaveLength(0);
+        expect(wrapper.find('.menu-icon')).toHaveLength(1);
+    });
+
+    it('shows the right menu items when the user is logged in', () => {
+        const wrapper = shallow(
+            <UserMenuBase
+                user={ { isAuthenticated: true } }
+                parameters={ { locale: 'locale', project: 'project' } }
+            />
+        );
+        wrapper.instance().setState({visible: true});
+
+        expect(wrapper.find('.details')).toHaveLength(1);
+        expect(wrapper.find('a[href="/settings/"]')).toHaveLength(1);
+        expect(wrapper.find(SignOut)).toHaveLength(1);
```

Comment:
One thing you can use in tests is the l10n ids. That easily allows you to check if a certain string is exposed. Not that it is needed here, but I wanted to make sure you had that tool in your belt. :)

Reply:
Thanks!

However, I'm trying this and it fails:
```
expect(wrapper.find('[data-l10n-id="user-UserMenu--settings"]')).toHaveLength(1);
```

Reply:
Here's an example of how to do it: https://github.com/mozilla/testpilot/blob/master/frontend/test/app/util.js#L4

I haven't yet used that, and so I haven't imported those functions into Pontoon, but I guess we'll do it at some point in the future.